synapse/rest/client/v2_alpha/devices.py

```diff
@@ -135,11 +137,12 @@ def on_DELETE(self, request, device_id):
         if not authed:
             defer.returnValue((401, result))
 
-        requester = yield self.auth.get_user_by_req(request)
-        yield self.device_handler.delete_device(
-            requester.user.to_string(),
-            device_id,
-        )
+        # check that the UI auth matched the access token
+        user_id = result[constants.LoginType.PASSWORD]
+        if user_id != requester.user.to_string():
+            raise errors.AuthError(403, "Invalid auth")
```

Comment:
I... am slightly surprised that `check_auth` doesn't either return or accept a user_id. OTOH, seems to be what everywhere else does so lets not try and fix it in this PR